### ISSUE TITLE
Sync upload: Remaining size/files status in menubar

### ISSUE THREAD

**Issue by godly-devotion** [SamSchott/maestral#428]
**Is your feature request related to a problem? Please describe.**
I just might be overlooking where it is but when syncing many files, it would be helpful to know how many files (and total file size) is left to be synced. Especially before I bring my laptop out to where there is no wifi.

**Describe the solution you'd like**
It would be informative if the menu bar could display the current sync status with information such as how many files remaining (and total file size left to be uploaded) rather than just saying `Syncing...` in the menu bar.

I believe this is an older screenshot but you can see here that the official Dropbox client also showing that info at the bottom.

![image](https://user-images.githubusercontent.com/1341760/126853729-92332211-f5c1-4f3f-a1a5-0f31c98663e5.png)

**Comment by SamSchott** [comment]
So, it depends on the situation. For performance reasons, Maestral fetches remote changes in batches of ~400 files and will show progress as it downloads each batch without knowing how more is to come. This is should be sufficiently informative for most situations where the number of remote items to sync is < 400. Unfortunately, the initial sync typically involves a much larger number of files. Maestral will show its progress as it fetches remote changes, for example `Indexing 4530...` but has no way of showing how many remain.

For upload syncs, we are in full control and always show the full progress, for example `Syncing ↑ 6/240...`.

The GUI currently does not display sync progress for individual files but the CLI does provide a live view with the command `maestral activity`.

**Comment by SamSchott** [comment]
I agree that having a live view for individual files would also be nice for the GUI, I've just not come around to actually implementing it. Ideally, it should be available for both macOS and Linux. Gnome however seems to have very limited supported for showing rich interfaces as pop-ups from the status bar.

**Comment by godly-devotion** [comment]
> I agree that having a live view for individual files would also be nice for the GUI, I've just not come around to actually implementing it. Ideally, it should be available for both macOS and Linux. Gnome however seems to have very limited supported for showing rich interfaces as pop-ups from the status bar.

Agreed, cross platform can be hard. Perhaps we could start with implementation on macOS and a partial feasible solution on Gnome for starters? Anything related to sync upload status can be very helpful.

**Comment by SamSchott** [comment]
The code base, even the GUI, is currently written in a mostly platform-independent way and I am a bit reluctant to give this up because of the increased maintenance work. But I'm afraid that there won't be away around that in the medium term, especially if some features are very difficult to implement for Linux.

> Especially before I bring my laptop out to where there is no wifi.

Yeah, that's a good use case. In the meantime, unless you need a file to be synced before you leave, feel free to interrupt a running sync at any time, either by pausing, quitting, or disconnecting from the internet. It will just pick up later from where you left off.
